A4A: Configure A4A Help center.

jkguidaven opened the pull request:
This PR updates the A4A Help Center to use the `automattic-chat-support_a4a` chat bot and moves the button to the Sidebar's footer next to the profile button.

<img width="311" height="456" alt="Screenshot 2025-11-14 at 7 42 27 PM" src="https://github.com/user-attachments/assets/a742eba6-e835-4f54-807d-5302946ad08e" />

Closes https://linear.app/a8c/issue/A4A-1805/a4a-move-helpcenter-icon-at-the-sidebar-footer

## Proposed Changes

* Reposition the Help center button at the sidebar's footer based on the design. MYGhJfDKwa8VdLu6WfFd7b-fi-12469_2263
* Set Help Center Odie bot to `automattic-chat-support_a4a` in the loader when in A4A environment.

## Why are these changes being made?

* This is part of the Cohesive support in the A4A project.

## Testing Instructions


* Use the A4A live link and navigate to `/overview` page.
* Confirm that the Help center button is located at the Sidebar's footer next to the profile menu button.
* Click the Help Center button to reveal the Help center.
* Click the `Get in touch` button.
* Asked for any A4A related questions.
* Confirm that the Bot is able to respond to the question.
    <img width="445" height="845" alt="Screenshot 2025-11-14 at 7 42 55 PM" src="https://github.com/user-attachments/assets/07da9969-7fe7-4799-a70e-ad17acb8e8ac" />


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

jkguidaven committed:
Set Odie bot to use 'automattic-chat-support_a4a'.

jkguidaven committed:
Remove unused useEffect dependency.

jkguidaven committed:
Set Odie bot in the loader.

jkguidaven committed:
Reposition Help Center button at the Sidebar footer.

alshakero commented on client/layout/help-center-loader.tsx:
```diff
@@ -54,6 +55,7 @@ export default function HelpCenterLoader( { sectionName, loadHelpCenter, current
 			hidden={ sectionName === 'gutenberg-editor' && isDesktop }
 			onboardingUrl={ onboardingUrl() }
 			googleMailServiceFamily={ getGoogleMailServiceFamily() }
+			newInteractionsBotSlug={ isA8CForAgencies() ? 'automattic-chat-support_a4a' : undefined }
```
This may cause issues, because the undefined will overwrite the default value because the prop is present. Could you do [like so](https://github.com/Automattic/wp-calypso/blob/trunk/apps/help-center/help-center-gutenberg.js#L172) instead?

jkguidaven replied:
Thanks for bringing this up. I initially thought setting the value to undefined would load the default, but it has now been fixed 👍

alshakero reviewed (commented):
Looks great. Left one comment.

yashwin commented on client/a8c-for-agencies/components/sidebar/header/profile-dropdown.tsx:
```diff
@@ -172,6 +177,8 @@ const ProfileDropdown = ( { compact, dropdownPosition = 'down' }: ProfileDropdow
 				) }
 			</Button>
+
+			{ withHelpCenter && <SidebarHelpCenter onClick={ () => {} } /> }
```
Can we remove or make the OnClick optional on the `SidebarHelpCenter` instead of doing it?

jkguidaven replied:
This has been resolved. 👍

yashwin reviewed (commented):
The changes mostly look good.

However, I left some comments. Please let me know your thoughts!

A few things I noticed:

Should we remove the border? I feel this looks neater:

CC: @madebynoam 

| Without Border | With Border |
| ------------- | ------------- |
| <img width="266" height="858" alt="Screenshot 2025-11-21 at 9 29 52 AM" src="https://github.com/user-attachments/assets/4f83c9ae-26be-4855-8599-e38eca375fe7" /> | <img width="266" height="858" alt="Screenshot 2025-11-21 at 9 29 58 AM" src="https://github.com/user-attachments/assets/05d7343a-ac04-434b-a963-25b89d209812" /> |



Also, I'm getting the same response for all the questions. I think we can ignore it for this PR.:

<img width="407" height="452" alt="Screenshot 2025-11-21 at 9 24 13 AM" src="https://github.com/user-attachments/assets/71f7e884-c800-44f5-833d-2c565cd6a939" />

jkguidaven committed:
Address PR comments.

jkguidaven committed:
Remove empty onClick handler.

jkguidaven commented:
I have addressed most of the comments from this PR. I'll go ahead and merge it. If there are additional feedback, we will handle them later during CFT.
